ksql-engine/src/main/java/io/confluent/ksql/util/KafkaTopicClientImpl.java

```diff
@@ -74,27 +76,40 @@ public boolean isTopicExists(String topic) {
 
   public Set<String> listTopicNames() {
     try {
-      AdminClient adminClient = AdminClient.create(adminClientConfig);
-      Set<String> topicNames =  adminClient.listTopics().names().get();
-      adminClient.close();
-      return topicNames;
+      return adminClient.listTopics().names().get();
     } catch (InterruptedException | ExecutionException e) {
       throw new KafkaResponseGetFailedException("Failed to retrieve kafka topic names", e);
     }
   }
 
   public Map<String, TopicDescription> describeTopics(Collection<String> topicNames) {
     try {
-      AdminClient adminClient = AdminClient.create(adminClientConfig);
-      Map<String, TopicDescription> topicInfos =  adminClient.describeTopics(topicNames).all()
+      return adminClient.describeTopics(topicNames).all()
           .get();
-      adminClient.close();
-      return topicInfos;
     } catch (InterruptedException | ExecutionException e) {
       throw new KafkaResponseGetFailedException("Failed to describe kafka topics", e);
     }
   }
 
+  public void deleteTopics(List<String> topicsToDelete) {
+    boolean hasDeleteErrors = false;
+    final DeleteTopicsResult deleteTopicsResult = adminClient.deleteTopics(topicsToDelete);
+    final Map<String, KafkaFuture<Void>> results = deleteTopicsResult.values();
+
+    for (final Map.Entry<String, KafkaFuture<Void>> entry : results.entrySet()) {
+      try {
+        entry.getValue().get(30, TimeUnit.SECONDS);
+      } catch (Exception e) {
+        System.err.println("ERROR: deleting topic " + entry.getKey());
+        e.printStackTrace(System.err);
+        hasDeleteErrors = true;
+      }
+    }
+    if (hasDeleteErrors) {
```

Comment:
rather than having a flag it would be better to collect all the failures into a list and then throw an exception with all the topics that failed. Also, we shouldn't be throwing `RuntimeException` - at the very least it should be `KsqlException`, but perhaps a `DeleteTopicsFailedException` or something.

Reply:
Good idea, Done!